tests/languages/json-source-code.test.js

```diff
@@ -497,4 +497,302 @@ describe("JSONSourceCode", () => {
+			const firstToken = sourceCode.ast.tokens[0];
+			assert.strictEqual(sourceCode.getTokenBefore(firstToken), null);
+		});
+
+		it("should return the previous token for a middle token", () => {
+			const tokens = sourceCode.ast.tokens;
+			const secondToken = tokens[1];
+			assert.strictEqual(
+				sourceCode.getTokenBefore(secondToken),
+				tokens[0],
+			);
+		});
+
+		it("should return null when passing a node that does not exist", () => {
+			assert.strictEqual(
+				sourceCode.getTokenBefore({ type: "String", range: [20, 25] }),
+				null,
+			);
+		});
+
+		it("should return the previous token when passing a node as the argument", () => {
+			const propertyNode = sourceCode.ast.body.members[0];
+			const valueNode = propertyNode.value;
+			const valueToken = sourceCode.ast.tokens.find(
+				token => token.range[0] === valueNode.range[0],
+			);
+			const prevToken = sourceCode.getTokenBefore(valueNode);
+			assert.strictEqual(
+				prevToken,
+				sourceCode.ast.tokens[
+					sourceCode.ast.tokens.indexOf(valueToken) - 1
+				],
+			);
+		});
+
+		it("should return null for a node at the start of the file", () => {
+			assert.strictEqual(sourceCode.getTokenBefore(sourceCode.ast), null);
+		});
+
+		it("should return the previous comment when includeComments is true", () => {
+			const commentFile = {
+				body: '{ "a" // comment\n: 1 }',
+				path: "test.jsonc",
+			};
+			const commentLanguage = new JSONLanguage({ mode: "jsonc" });
+			const commentParseResult = commentLanguage.parse(commentFile);
+			const commentSourceCode = new JSONSourceCode({
+				text: commentFile.body,
+				ast: commentParseResult.ast,
+			});
+
+			const tokens = commentSourceCode.ast.tokens;
+			const colonToken = tokens.find(token => token.type === "Colon");
+			assert.strictEqual(
+				commentSourceCode.getTokenBefore(colonToken),
+				tokens[1],
+			);
+		});
+	});
+
+	describe("getTokenAfter()", () => {
+		let sourceCode;
+		let file;
+		let language;
+		let parseResult;
+
+		beforeEach(() => {
+			file = { body: '{"foo": 123}', path: "test.json" };
+			language = new JSONLanguage({ mode: "json" });
+			parseResult = language.parse(file);
+			sourceCode = new JSONSourceCode({
+				text: file.body,
+				ast: parseResult.ast,
+			});
+		});
+
+		it("should return the next token after a node", () => {
+			const stringNode = parseResult.ast.body.members[0].name;
+			const nextToken = sourceCode.getTokenAfter(stringNode);
+
+			assert.strictEqual(nextToken.type, "Colon");
+		});
+
+		it("should return the next token after a token", () => {
+			const openBraceToken = parseResult.ast.tokens.find(
+				token => token.type === "LBrace",
+			);
+			const nextToken = sourceCode.getTokenAfter(openBraceToken);
+
+			assert.strictEqual(nextToken.type, "String");
+		});
+
+		it("should skip comments when getting next token", () => {
+			const commentFile = {
+				body: '{\n// comment\n"foo": true}',
+				path: "test.json",
```

Comment:
```suggestion
				path: "test.jsonc",
```

I think `jsonc` is more appropriate here, since the `body` uses comment.